### PR TITLE
Refactor dubbo spring boot structure

### PR DESCRIPTION
**Pull request by CrazyHZM**
## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test to verify your logic correction. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [x] Make sure gitHub actions can pass. [Why the workflow is failing and how to fix it?](./CONTRIBUTING.md)

### CODE REVIEW

**Review by AlbumenJ — APPROVED** [review]

**Comment by oxsean** [comment]
### Some suggestions for reference only:
- Rename `dubbo-spring-boot-project` to `dubbo-spring-boot-all`
- Move `dubbo-spring-boot-compatible/dubbo-spring-boot-autoconfigure-compatible` to `dubbo-spring-boot-1-autoconfigure`
- Move `dubbo-spring-boot-compatible/dubbo-spring-boot-actuator-autoconfigure-compatible` to `dubbo-spring-boot-1-actuator-autoconfigure`

File: pom.xml [hunk 1]
```diff
@@ -89,9 +89,15 @@
     <module>dubbo-dependencies</module>
     <module>dubbo-metadata</module>
     <module>dubbo-metrics</module>
-    <module>dubbo-spring-boot</module>
     <module>dubbo-test</module>
     <module>dubbo-maven-plugin</module>
+    <module>dubbo-spring-boot-project/dubbo-spring-boot-actuator</module>
+    <module>dubbo-spring-boot-project/dubbo-spring-boot-autoconfigure</module>
+    <module>dubbo-spring-boot-project/dubbo-spring-boot-compatible</module>
+    <module>dubbo-spring-boot-project/dubbo-spring-boot-3-autoconfigure</module>
```
Comment (oxsean) [inline pom.xml]: Should move to profile `jdk-version-ge-17`

File: dubbo-spring-boot/pom.xml [hunk 2]
```diff
@@ -1,256 +0,0 @@
...
-              <Specification-Version>${project.version}</Specification-Version>
-              <Implementation-Version>${project.version}</Implementation-Version>
-            </manifestEntries>
-          </archive>
-        </configuration>
-      </plugin>
-    </plugins>
-  </build>
-
-  <profiles>
```
Comment (oxsean) [inline dubbo-spring-boot/pom.xml:214]: I think these profiles are still useful and need to be kept

File: dubbo-spring-boot-project/dubbo-spring-boot-autoconfigure/pom.xml
Comment (oxsean) [inline dubbo-spring-boot-project/dubbo-spring-boot-autoconfigure/pom.xml:1]: `dubbo-config-spring` appeared twice

File: dubbo-spring-boot-project/dubbo-spring-boot-3-autoconfigure/pom.xml
Comment (oxsean) [inline dubbo-spring-boot-project/dubbo-spring-boot-3-autoconfigure/pom.xml:1]: Add a `<spring-boot-3.version>3.2.1</spring-boot-3.version>` to bom and use `<version>${spring-boot-3.version}</version>` ?

File: dubbo-spring-boot/pom.xml (hunk 2 above)
Comment (CrazyHZM) [reply dubbo-spring-boot/pom.xml:214]: Some of the required profiles have been moved to the parent pom.

File: dubbo-spring-boot-project/dubbo-spring-boot-autoconfigure/pom.xml
Comment (CrazyHZM) [reply dubbo-spring-boot-project/dubbo-spring-boot-autoconfigure/pom.xml:1]: Done.

File: dubbo-spring-boot-project/dubbo-spring-boot-3-autoconfigure/pom.xml
Comment (CrazyHZM) [reply dubbo-spring-boot-project/dubbo-spring-boot-3-autoconfigure/pom.xml:1]: dependencyManagement needs to be added to the submodule. Changing the properties name does not have much effect.

File: pom.xml (hunk 1 above)
Comment (CrazyHZM) [reply pom.xml]: Have already existed.

**Comment by CrazyHZM** [comment]
> ### Some suggestions for reference only:
> * Rename `dubbo-spring-boot-project` to `dubbo-spring-boot-all`
> * Move `dubbo-spring-boot-compatible/dubbo-spring-boot-autoconfigure-compatible` to `dubbo-spring-boot-1-autoconfigure`
> * Move `dubbo-spring-boot-compatible/dubbo-spring-boot-actuator-autoconfigure-compatible` to `dubbo-spring-boot-1-actuator-autoconfigure`

`dubbo-spring-boot-all` strikes me as fatjar, but for now, it's just a directory of files, so I think d`ubbo-spring-boot-project` is more appropriate

`dubbo-spring-boot-compatible `also needs to exist because it is managed as a 1.x bom, otherwise, it cannot override the parent's high version dependencies.

**Review by oxsean — APPROVED** [review]

**Pull request merged**